File: day01/python/part1.py

def main(file_path: str):
    sum = 0
    with open(file_path) as file:
        for line in file:
            nums = []
            for char in line:
                if char.isdigit():
                    nums.append(char)
            print(nums)
            sum += int(str(nums[0]) + str(nums[-1]))
    return sum


if __name__ == '__main__':
    print(main('input.txt'))

File: day01/go/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"unicode"
)

func check(e error) {
	if e != nil {
		fmt.Println("Error:", e)
		os.Exit(1)
	}
}

func main() {
	file, err := os.Open("files/input.txt")
	check(err)
	defer file.Close()
	scanner := bufio.NewScanner(file)
	sum := 0
	for scanner.Scan() {
		curr := ""
		var line = scanner.Text()
		for _, char := range line {
			if !unicode.IsLetter(char) {
				curr += string(char)
				break
			}
		}
		var line_rune = []rune(line)
		for i := len(line_rune) - 1; i >= 0; i-- {
			char := line_rune[i]
			if !unicode.IsLetter(char) {
				curr += string(char)
				break
			}
		}
		curr_int, _ := strconv.Atoi(curr)
		sum += curr_int
	}
	fmt.Println(sum)
}
